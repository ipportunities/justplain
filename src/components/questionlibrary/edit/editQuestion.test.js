import React from 'react'
import { shallow } from 'enzyme'
import EditQuestion from './editQuestion'

const questionMock = {
  id: 1,
  question: 'Some question to ask?',
  type: 'OPEN',
  subject: 'subject',
  difficulty: 'EASY',
  tags: ['tag', 'tog', 'tig'],
  newTag: '',
  answers: [
    {
      id: 1,
      answer: 'A',
      correct: true,
      order: 0
    }
  ],
  editing: false
}

const saveQuestionMock = jest.fn()
const removeQuestionMock = jest.fn()

jest.mock('react-redux', () => ({
  ...jest.requireActual('react-redux'),
  useSelector: jest.fn().mockReturnValue({ auth: { token: 'a' } })
}))

describe('Edit question', () => {
  it('should render correctly', () => {
    const component = shallow(<EditQuestion question={ questionMock } saveQuestion={ saveQuestionMock }
                                            removeQuestion={ removeQuestionMock } removeLoading={ false } />)
    expect(component).toMatchSnapshot()
  })
})
