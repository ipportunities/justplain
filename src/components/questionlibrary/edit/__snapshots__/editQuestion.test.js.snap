// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Edit question should render correctly 1`] = `
<div
  className="col-8 offset-2 p-2"
>
  <div
    className="row"
  >
    <div
      className="col-10"
    >
      <div
        className="form-group"
      >
        <label
          htmlFor="question-1"
        >
          Vraag
        </label>
        <input
          className="form-control pl-2 pr-2"
          disabled={true}
          id="question-1"
          name="question"
          onChange={[Function]}
          type="text"
          value="Some question to ask?"
        />
      </div>
    </div>
    <div
      className="col-2 btn-group"
    >
      <button
        className="btn btn-outline-primary mt-4"
        onClick={[Function]}
      >
        Edit
      </button>
      <button
        className="btn btn-outline-danger mt-4"
        disabled={false}
        onClick={[Function]}
      >
        Remove
      </button>
    </div>
  </div>
</div>
`;
